larq_compute_engine/mlir/ir/lce_ops.td

```diff
@@ -35,25 +35,21 @@ TODO
   }];
 
   let arguments = (ins
-    TensorOf<[BF16, F16, F32, F64, I32]>:$input,
-    TensorOf<[BF16, F16, F32, F64, I32]>:$filter,
+    TensorOf<[F32]>:$input,
+    TensorOf<[F32]>:$filter,
     TensorOf<[F32]>:$fused_multiply,
     TensorOf<[F32]>:$fused_add,
 
     I64ArrayAttr:$strides,
-    TF_AnyStrAttrOf<["SAME", "VALID", "EXPLICIT"]>:$padding,
+    TF_AnyStrAttrOf<["SAME", "VALID"]>:$padding,
     DefaultValuedAttr<TF_ConvnetDataFormatAttr, "NHWC">:$data_format,
     DefaultValuedAttr<I64ArrayAttr, "{1, 1, 1, 1}">:$dilations,
-    DefaultValuedAttr<TF_AnyStrAttrOf<["HWIO", "OHWI", "OHWI_PACKED"]>, "HWIO">:$filter_format
+    DefaultValuedAttr<TF_AnyStrAttrOf<["HWIO", "OHWI", "OHWI_PACKED"]>, "OHWI">:$filter_format
```

Comment:
```suggestion
    DefaultValuedAttr<TF_AnyStrAttrOf<["OHWI", "OHWI_PACKED"]>, "OHWI">:$filter_format
```

Reply:
but we still support it in the RefImpl. right? we are planning to remove the RefImpl. (it is not used anyway) but its not remove *yet* so maybe just remove it with the same PR which removes the RefImpl.?

Reply:
Currently in `Prepare` we already assert that its OHWI, [see this line](https://github.com/larq/compute-engine/blob/1071647e0b045e6b9fe5666a55a367cca6aff99e/larq_compute_engine/tflite/kernels/bconv2d.cc#L175) .
And the RefImp also assumes OHWI now (although it probably still has some code for HWIO support somewhere).

Reply:
but RefImpl do not support OHWI format? so if someone just go and change to the ref. impl. for lets say only testing purposes the code will not work! so the current code is already inconsistent with itself and we should not have that assert at the first place. I am gonna send a PR to cleanup this mess.

Reply:
The Ref Impl also has never supported dilations for example. I think we should just remove it asap.